src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -133,20 +138,28 @@ public void onConnectionResumed(boolean sessionPresent) {
      * @param executorService {@link ExecutorService}
      * @param deviceConfiguration {@link DeviceConfiguration}
      * @param iotShadowClient {@link IotShadowClient}
+     * @param kernel {@link Kernel}
      */
     public ShadowDeploymentListener(DeploymentQueue deploymentQueue, DeploymentStatusKeeper deploymentStatusKeeper,
                                     MqttClient mqttClient, ExecutorService executorService,
-                                    DeviceConfiguration deviceConfiguration, IotShadowClient iotShadowClient) {
+                                    DeviceConfiguration deviceConfiguration, IotShadowClient iotShadowClient,
+                                    Kernel kernel) {
         this.deploymentQueue = deploymentQueue;
         this.deploymentStatusKeeper = deploymentStatusKeeper;
         this.mqttClient = mqttClient;
         this.executorService = executorService;
         this.deviceConfiguration = deviceConfiguration;
         this.iotShadowClient = iotShadowClient;
+        this.kernel = kernel;
     }
 
     @Override
     public void postInject() {
+        if (iotShadowClient == null) {
```

Comment:
why does this need to be in postInject?  shouldn't the shadow client be reconfigured when         deviceConfiguration changes?

Reply:
The connection may need to be re-established upon device configuration changes. That is take care by the injected MqttClient object. For IotShadowClient it is abstracted.